From 28d54874671461ac19a381fff51293ac77a5953b Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Tue, 2 Jul 2024 14:12:49 +0200
Subject: [PATCH 16/36] Revert "demux: mkv: fix seek segfault with invalid
 chapters"

This reverts commit 4ba01f2831193b2a59513c897b159c14c0a9f086.

Let p_vsegment->Seek() decide what is valid or not.
The p_segment is actually the p_vsegment's current chapter's actual segment.
The current chapter segment may be NULL, but that doesn't mean we can't seek.
The current chapter segment cannot be NULL, per its constructor.

(cherry picked from commit 298c59f44b6f14173847ba0108725a55681e0794)
---
 modules/demux/mkv/mkv.cpp | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/modules/demux/mkv/mkv.cpp b/modules/demux/mkv/mkv.cpp
index 90a5bd5bc5..6bdaeb3e39 100644
--- a/modules/demux/mkv/mkv.cpp
+++ b/modules/demux/mkv/mkv.cpp
@@ -452,7 +452,6 @@ static int Seek( demux_t *p_demux, vlc_tick_t i_mk_date, double f_percent, virtu
 {
     demux_sys_t        *p_sys = p_demux->p_sys;
     virtual_segment_c  *p_vsegment = p_sys->p_current_vsegment;
-    matroska_segment_c *p_segment = p_vsegment->CurrentSegment();
 
     if( f_percent < 0 ) msg_Dbg( p_demux, "seek request to i_pos = %" PRId64, i_mk_date );
     else                msg_Dbg( p_demux, "seek request to %.2f%%", f_percent * 100 );
@@ -472,11 +471,6 @@ static int Seek( demux_t *p_demux, vlc_tick_t i_mk_date, double f_percent, virtu
         msg_Warn( p_demux, "cannot seek without duration!");
         return VLC_EGENERIC;
     }
-    if( !p_segment )
-    {
-        msg_Warn( p_demux, "cannot seek without valid segment position");
-        return VLC_EGENERIC;
-    }
 
     /* seek without index or without date */
     if( f_percent >= 0 && (var_InheritBool( p_demux, "mkv-seek-percent" ) || i_mk_date < 0 ))
-- 
2.55.0

